[Edit] C++ Math Functions: `exp()` (#1033)
Codecademy/docs: content/cpp/concepts/math-functions/terms/exp/exp.md

---
Title: 'exp()'
Description: 'Returns the base-e exponential function of the argument.'
Subjects:
  - 'Computer Science'
Tags:
  - 'Functions'
  - 'Arithmetic'
  - 'Methods'
CatalogContent:
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
---

The **exp()** function returns the base-_e_ exponential value of the argument (i.e., _e_ raised to the power of the argument). This is the inverse of the [`log()`](https://www.codecademy.com/resources/docs/cpp/math-functions/log) function.

## Syntax

> ```cpp

Reviewer: ```suggestion
```pseudo
```

The \```pseudo code snippet is commonly used in the "Syntax" section of an entry.

---
Title: 'exp()'
Description: 'Returns the base-e exponential function of the argument.'
Subjects:
  - 'Computer Science'
Tags:
  - 'Functions'
  - 'Arithmetic'
  - 'Methods'
CatalogContent:
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
---

> The **exp()** function returns the base-_e_ exponential function of the argument. This is _e_ raised to the power of the argument. This is the inverse of the [`log()`](https://www.codecademy.com/resources/docs/cpp/math-functions/log) function.

Reviewer: ```suggestion
The **exp()** function returns the base-_e_ exponential value of the argument (i.e., _e_ raised to the power of the argument). This is the inverse of the [`log()`](https://www.codecademy.com/resources/docs/cpp/math-functions/log) function.
```

## Syntax

```cpp
 exp(x)
```

> The `x` parameter can be any value, positive, negative or zero.

Reviewer: ```suggestion
The `x` parameter is a `double` value that can be positive, negative, or zero.
```

## Example

> The following example uses the `exp()` function to find the exponential function of `3`:

Reviewer: ```suggestion
The following example uses the `exp()` function to find the value of e<sup>3</sup>:
```

Author: I will use this suggestion for the section devoted to the Codebyte example.

```diff
-The following example is runnable and returns the exponential function of `1`:
+The following example is runnable and returns the value of e<sup>1</sup>:
```

```cpp
#include <iostream>
#include <math.h>

int main() {
  double x = 3;
  double result;
  result = exp(x);
  std::cout << "The exp of " << x << " is " << result << "\n";

>   // Output: The exp of 3 is 20.0855
> }
> ```
> 

Reviewer: ```suggestion
}

This produces the following output:

```

And then add a new \```shell snippet featuring the output:

```shell
The exp of 3 is 20.0855
```

## Codebyte Example

The following example is runnable and returns the exponential function of `1`:

```codebyte/cpp
#include <iostream>

> #include <math.h>

Reviewer: Not sure about this but isn't `math.h` more relevant to C programs as opposed to C++ programs? cc: @SSwiniarski

Author: Apologies, I looked at the `log` function and it used `<math.h>`. I guess you could raise an issue to update the [log](https://www.codecademy.com/resources/docs/cpp/math-functions/log) entry as well.
